### PR TITLE
Moving spell name translator call into a virtual method

### PR DESCRIPTION
**Pull request by Elec0**
I know you said you were going to do it, but I did it first. :p

### CODE REVIEW

**Comment by spacechase0** [comment]
I think it'd make more sense if, instead of a `getTranslation(key)` function that just redirected to the mod's translation helper, there was a `getTranslatedName()` and `getTranslatedDescription()`.

File: Magic.cs [hunk 1]
```diff
@@ -166,7 +166,7 @@ private static void onAnalyze(object sender, AnalyzeEventArgs e)
                     Log.debug("Player learnt spell: " + spell);
                     farmer.learnSpell(spell, 0, true);
                     //Game1.drawObjectDialogue(Mod.instance.Helper.Translation.Get("spell.learn", new { spellName = Mod.instance.Helper.Translation.Get("spell." + spell + ".name") }));
-                    Game1.addHUDMessage(new HUDMessage(Mod.instance.Helper.Translation.Get("spell.learn", new { spellName = Mod.instance.Helper.Translation.Get("spell." + spell + ".name") })));
+                    Game1.addHUDMessage(new HUDMessage(Mod.instance.Helper.Translation.Get("spell.learn", new { spellName = SpellBook.get(spell).getTranslation("spell." + spell + ".name") })));
```
Comment (spacechase0) [inline Magic.cs]: Also, here, instead of using `SpellBook.get(spell)` you can just use `spell` directly.

Comment (Elec0) [reply Magic.cs]: Did you extend string somehow? Because `List<string> spellsLearnt = new List<string>();` is the list `spell` comes from.

**Comment by Elec0** [comment]
> I think it'd make more sense if, instead of a `getTranslation(key)` function that just redirected to the mod's translation helper, there was a `getTranslatedName()` and `getTranslatedDescription()`.

Fair enough. I switched the naming.

File: Magic.cs (hunk 1 above)
Comment (spacechase0) [reply Magic.cs]: Whoops, you're right. I misread.

**Pull request merged**